tests/test_retry_plugin.py

```diff
@@ -969,3 +974,54 @@ def test_moar_flaky() -> None:
     assert "\ttest_flaky passed on attempt 3!" in result.outlines
     assert "\ttest_moar_flaky failed on attempt 1! Retrying!" in result.outlines
     assert "\ttest_moar_flaky passed on attempt 2!" in result.outlines
+
+
+@xdist_test_marker
+def test_xdist_resources_properly_closed_server_side(testdir):
+    # TODO: This test works for the sockets opened in the main process,
+    #       but there is no way to catch them inside the workers
+    #       (or at least, the author of this test didn't find it)
+
+    testdir.makepyfile(
+        """
+        import pytest
+        import warnings
+
+        class MyWarning(Warning):
+            pass
+
+        a = 0
+        b = 0
+
+        def func_a_that_randomly_warns():
+            global a
+
+            a += 1
+            if a == 3:
+                warnings.warn("Test", category=MyWarning)
+
+        def func_b_that_randomly_warns():
+            global b
+
+            b += 1
+            if b == 2:
+                warnings.warn("Test", category=MyWarning)
+
+        def test_flaky_a():
+
+            with pytest.warns(MyWarning):
+                func_a_that_randomly_warns()
+
+        def test_flaky_b():
+
+            with pytest.warns(MyWarning):
+                func_b_that_randomly_warns()
+        """
+    )
+
+    # The test MUST be run in a subprocess because the warnings appears
+    # on pytest teardown
+    result = testdir.runpytest_subprocess("-n", "2", "--retries", "3", "-Werror")
```

Comment:
"-W" "error". Although fwiw, this isn't a necessary param since you're using the `pytest.warn()` context manager, so feel free to just remove it.

Reply:
The `-W` flag is necessary to force pytest to log the `ResourceWarning` if it appears, because with `-Wdefault`, `ResourceWarning` is ignored. But somehow it gets logged when `pytest.warns()` is used.